winlogbeat/sys/strings.go

```diff
@@ -44,3 +44,27 @@ func RemoveWindowsLineEndings(s string) string {
 	s = strings.Replace(s, "\r\n", "\n", -1)
 	return strings.TrimRight(s, "\n")
 }
+
+// BinaryToString converts a binary field which is encoded in hexadecimal
```

Comment:
Good point, I think the complexity of maintaining it is quite low in this case, and in systems where such events are common it might pay off. Some bench results show a signifficant improvement in ns/op:

```
goos: windows
goarch: amd64
pkg: github.com/elastic/beats/v7/winlogbeat/sys
cpu: 13th Gen Intel(R) Core(TM) i7-13700H
BenchmarkBinaryToString/BinaryToString-10                                       14376853            82.90 ns/op       96 B/op          2 allocs/op
BenchmarkBinaryToString/strings.ToUpper(hex.EncodeToString(bin))-10              5623237            186.7 ns/op      144 B/op          3 allocs/op
PASS
ok  	github.com/elastic/beats/v7/winlogbeat/sys	2.563s
```